src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +144,1336 @@ static DamageTable D_80BBF858 = {
+        this->snorePhase = 0x400 << (actorType + 1);
+        if (!(actorType % 2)) {
+            this->actionFlags |= EN_S_GORO_ACTIONFLAG_SNOREPHASE;
+        }
+        this->actionFlags |= EN_S_GORO_ACTIONFLAG_SUPPRESS_SNORE;
+        this->actionFunc = EnSGoro_SleepAction;
+        return true;
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBD8F0.s")
+void EnSGoro_UpdateToHandtapAnimation(EnSGoro* this) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBD93C.s")
+    s16 curFrame = this->skelAnime.curFrame;
+    s16 lastFrame = Animation_GetLastFrame(sAnimationInfo[this->animInfoIndex].animation);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBD98C.s")
+    if ((this->animInfoIndex != EN_S_GORO_ANIM_STAND_HANDTAP) && (curFrame == lastFrame)) {
+        this->animInfoIndex = EN_S_GORO_ANIM_STAND_HANDTAP;
+        SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBDACC.s")
+void EnSGoro_UpdateSleepyAnimation(EnSGoro* this) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBDC34.s")
+    s16 curFrame = this->skelAnime.curFrame;
+    s16 lastFrame = Animation_GetLastFrame(sAnimationInfo[this->animInfoIndex].animation);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBDCFC.s")
+    if (this->animInfoIndex != EN_S_GORO_ANIM_SLEEPY) {
+        if (curFrame == lastFrame) {
+            this->animInfoIndex = EN_S_GORO_ANIM_SLEEPY;
+            SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+        }
+    } else if (curFrame == lastFrame) {
+        this->actionFlags &= ~EN_S_GORO_ACTIONFLAG_TIRED;
+        this->animInfoIndex = EN_S_GORO_ANIM_IDLE_STAND;
+        SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBDDF8.s")
+void EnSGoro_UpdateToIdleAnimation(EnSGoro* this) {
+
+    s16 curFrame = this->skelAnime.curFrame;
+    s16 lastFrame = Animation_GetLastFrame(sAnimationInfo[this->animInfoIndex].animation);
+
+    if ((this->animInfoIndex != EN_S_GORO_ANIM_IDLE_STAND) && (curFrame == lastFrame)) {
+        this->animInfoIndex = EN_S_GORO_ANIM_IDLE_STAND;
+        SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+    }
+}
+
+void EnSGoro_UpdateCollider(EnSGoro* this, PlayState* play) {
+
+    Vec3f world_pos = this->actor.world.pos;
+    f32 radius = 24.0f;
+    f32 height = 62.0f;
+
+    this->collider.dim.pos.x = world_pos.x;
+    this->collider.dim.pos.y = world_pos.y;
+    this->collider.dim.pos.z = world_pos.z;
+    this->collider.dim.radius = radius;
+    this->collider.dim.height = height;
+
+    // It is not clear what this is for. May be a bug.
```

Comment:
```suggestion
    //! @bug: It is not clear what this is for.
```
Clearly marks as a bug, and makes it easier to find.